rust/lance-index/src/vector/hnsw/builder.rs

```diff
@@ -246,7 +246,7 @@ impl HNSWBuilderInner {
     /// See paper `Algorithm 1`
     fn random_level(&self) -> u16 {
         let mut rng = thread_rng();
-        let ml = self.params.m as f32;
+        let ml = 1.0 / (self.params.m as f32).ln();
```

Comment:
nit: could we add a comment to say "this is from https://arxiv.org/pdf/1603.09320.pdf section 4.1"

Reply:
line 246 has done that.
need to add the link?

Reply:
oh missed that part. lgtm